### PR TITLE
Reduce kSpake2p_Iteration_Count to fix timeouts

### LINKED ISSUE #3089: Calculating key with SHA256 on nrfconnect and nrf5 platforms takes very long time.
#### Problem
When initializing Rendezvous session there is a need to calculate key using pbkdf2_sha256() method. On the both nRF platforms - nrf5 and nrfconnect (as both are using mbedTLS library) it was observed that application is spending very long time on calculating this key, which is probably not acceptable. Time of calculations is strictly connected with the kSpake2p_Iteration_Count value, which stands for numbers of iterations performed in generating the key.

Board used in testing: nRF52840.

Observed conclusions:
* kSpake2p_Iteration_Count is constant value common for all platforms and equal to 50 000.
* Time spend on calculations is equal to over 50 s for 50 000 iterations at nrfconnect platform.
* Lowering kSpake2p_Iteration_Count value to 4096 leads to shortening calculation time to about 4,5 s, which is still far too long.
* Basing on analyzing configuration options it was checked that application is using hardware module for calculation, not the software one (which is desired and should fasten calculations). 

Questions to answer:
* What is the reason of spending such a long time on calculations?
* Do we need such a big number of iterations like 50 000? What is the background of selecting that value? It obviously increases the security but isn't this value far too big for the considered use case?

 #### Proposed Solution
Lowering the number of iterations may shorten the calculation time, but is not solving problem at all, so should be done only if the security requirements allow to this.

### PR DESCRIPTION
**Pull request by mspang**
Initializing this takes 15 seconds on ESP32, which is longer than most
timers we specify (e.g. BLE disconnect timeout is 5 seconds).

Reduce this to a value that doesn't interfere with other work.

fixes #3089

 #### Problem

Spake2p initialization causes timeouts.

 #### Summary of Changes

Reduce iterations by 100x.

### CODE REVIEW

**Review by rwalker-apple — DISMISSED** [review]
reference: https://tools.ietf.org/html/rfc2898

**Comment by rwalker-apple** [comment]
catching up on this. can we cover the value of this in a compile-time flag?  will it present a problem if a software implementation ships and uses this low value?

**Comment by mspang** [comment]
> catching up on this. can we cover the value of this in a compile-time flag? will it present a problem if a software implementation ships and uses this low value?

I don't think it's a good idea for this to be configurable, different choices are incompatible.

File: src/transport/RendezvousSession.cpp [hunk 1]
```diff
@@ -28,7 +28,7 @@
 #endif // CONFIG_NETWORK_LAYER_BLE
 
 static const size_t kMax_SecureSDU_Length          = 1024;
-static constexpr uint32_t kSpake2p_Iteration_Count = 50000;
+static constexpr uint32_t kSpake2p_Iteration_Count = 500;
```
[the remark this answers is not in the document]
Comment (mspang) [reply src/transport/RendezvousSession.cpp]: Done. I had that originally but apparently 1000 is still going to take 1 second on nrf52840-dk which still seemed costly to me. And this code is apparently temporary anyway.

**Comment by mspang** [comment]
> > catching up on this. can we cover the value of this in a compile-time flag? will it present a problem if a software implementation ships and uses this low value?

Yes, the handshake will fail so the product will not interoperate with other devices.

> 
> I don't think it's a good idea for this to be configurable, different choices are incompatible.

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Comment by rwalker-apple** [comment]
it occurs to me that if we can't ship with this low of a value anyway, why not set it to zero or one?

**Review by rwalker-apple — APPROVED** [review]

**Comment by rwalker-apple** [comment]
@mspang, feel free to land this yourself unless you want to lower the value more

**Comment by mspang** [comment]
> it occurs to me that if we can't ship with this low of a value anyway, why not set it to zero or one?

That was my thinking with 500. I lowered it to 100, I'm not sure if 0 is legal and 1 seems a bit extreme.

**Pull request merged**